### ISSUE TITLE
`InsertOrReplace` ignores `[AutoIncrement]` on keys (inserts with `0` value)

### ISSUE THREAD

**Issue by justdmitry** [praeclarum/sqlite-net#1091]
Code to reproduce:

```csharp
using SQLite;

var db = new SQLiteConnection("test.sqlite");

db.CreateTable<Item>();

db.InsertOrReplace(new Item { Value = 10 });
db.InsertOrReplace(new Item { Value = 20 });
Console.WriteLine(db.Table<Item>().Count()); // prints "1"

db.DeleteAll<Item>();
db.Insert(new Item { Value = 10 });
db.Insert(new Item { Value = 20 });
Console.WriteLine(db.Table<Item>().Count()); // prints "2"

db.DropTable<Item>();


public class Item
{
    [PrimaryKey, AutoIncrement]
    public int Id { get; set; }

    public int Value { get; set; }
}

```

After `InsertOrReplace` table contains only one row with Id=0 and Value=20.
It seems that `AutoIncrement` is ignored and default Id=0 is used.

**sqlite-net-pcl v1.8.116**, .net6.0, windows 10

**Comment by AlexITMS** [comment]
try changing the 

public int Id {get; set;}

to 

public int Id? [get; set;}

**Comment by justdmitry** [comment]
I guess not to `public int Id?` but to `public int? Id` (question mark after datatype, not after name).

Yes, this works, thank you. But this adds warning in all other usages of class, so I'll keep calling `Insert` or `Update` as long as I know if record exist.

I still think original (`int Id`) should work consistent between Insert and InsertOrUpdate and either both use `0` as valid value, or replace it with autoincremented one.

**Comment by AlexITMS** [comment]
yes sorry typo on my part.

**Comment by Hottemax** [comment]
> I guess not `public int Id?` but to `public int? Id` (question mark after datatype, not after name).
> 
> Yes, this works, thank you. But this adds warning in all other usages of class, so I'll keep calling `Insert` or `Update` as long as I know if record exist.
> 
> I still think original (`int Id`) should work consistent between Insert and InsertOrUpdate and either both use `0` as valid value, or replace it with autoincremented one.

@justdmitry I do not agree, the current behavior is correct in my opinion and how I would expect `InsertOrReplace` should operate.
Your `new Item { Value = <someint> }` is identical to `new Item { Id = 0, Value = <someint> }`, hence `InsertOrReplace` will replace the item with `Id=0` as one would expect.

**Comment by justdmitry** [comment]
@Hottemax
If `Id = 0` is "correct value, that should be saved to database" then why `Insert` thinks this is "invalid" value and applies autoincrement logic to it?

With your logic, first `Insert` should add record with Id=0 into DB, ans second `Insert` should throw exception "Key already exist".

In my opinion, same value (Id=0) should be either used as "valid" by all methods (and saved into DB unchanged), or be replaced with autoincremented one by both.

**Comment by NetworkGithub** [comment]
See #327

**Comment by Hottemax** [comment]
> @Hottemax If `Id = 0` is "correct value, that should be saved to database" then why `Insert` thinks this is "invalid" value and applies autoincrement logic to it?
> 
> With your logic, first `Insert` should add record with Id=0 into DB, ans second `Insert` should throw exception "Key already exist".
> 
> In my opinion, same value (Id=0) should be either used as "valid" by all methods (and saved into DB unchanged), or be replaced with autoincremented one by both.

@justdmitry You are correct, I was mistaken. I guess using AutoIncrement and InsertOrReplace is not quite intuitive to the human brain, which might interpret it differently ("what gets resolved first" etc.). But it is not consistent as it is, you are right.

**Comment by justdmitry** [comment]
Thank you @NetworkGithub . I think this issue may be closed as duplicate of #327.
